Source/NSCollectionView.m

```diff
@@ -1114,4 +1456,558 @@ - (BOOL) wantsPeriodicDraggingUpdates
+}
+
+- (BOOL) backgroundViewScrollsWithContent
+{
+  return _backgroundViewScrollsWithContent;
+}
+
+- (void) setBackgroundViewScrollsWithContent: (BOOL)f
+{
+  _backgroundViewScrollsWithContent = f;
+}
+
+/* Reloading Content */
+- (void) _loadItemAtIndexPath: (NSIndexPath *)path
+{
+  NSCollectionViewItem *item = [self makeItemWithIdentifier: nil
+                                               forIndexPath: path];
+  
+  if (item != nil)
+    {
+      NSView *v = [item view];
+      NSRect f = [v frame];
+      _GSCollectionViewItemTrackingView *tv =
+	[[_GSCollectionViewItemTrackingView alloc]
+	  initWithFrame: NSMakeRect(0.0, 0.0, f.size.width, f.size.height)];
+
+      // Set up tracking view...
+      [v setNextResponder: tv];
+      [tv setAlphaValue: 0.0];
+      [tv setItem: item];
+      [tv setCollectionView: self];
+      [tv setIndexPath: path];
+      [v addSubview: tv positioned: NSWindowAbove relativeTo: nil];
+      RELEASE(tv);
+      
+      [_visibleItems addObject: item];
+      [_indexPathsForVisibleItems addObject: path];
+      if (_collectionViewLayout)
+        {
+          NSCollectionViewLayoutAttributes *attrs =
+            [self layoutAttributesForItemAtIndexPath: path];
+          NSRect frame = [attrs frame];
+          BOOL hidden = [attrs isHidden];
+          CGFloat alpha = [attrs alpha];
+          NSSize sz = [attrs size];
+          
+	  // set attributes of item based on currently selected layout...
+          frame.size = sz;
+          [v setFrame: frame];
+          [v setHidden: hidden];
+          [v setAlphaValue: alpha];
+
+          [_itemsToAttributes setObject: attrs
+                                 forKey: item];
+          
+          [self addSubview: v];
+        }
+      else
+        {
+          NSLog(@"NSCollectionViewLayout subclass is not set");
+        }
+    }
+}
+
+- (void) _loadSectionAtIndex: (NSUInteger)cs
+{
+  NSInteger ni = [self numberOfItemsInSection: cs];
+  NSInteger ci = 0;
+  
+  for (ci = 0; ci < ni; ci++)
+    {
+      NSIndexPath *p = [NSIndexPath indexPathForItem: ci inSection: cs];
+      [self _loadItemAtIndexPath: p];
+    }
+}
+
+- (void) _clearMaps
+{
+  // Remove objects from set/dict/maps
+  [_visibleItems removeAllObjects];
+  [_indexPathsForVisibleItems removeAllObjects];
+  [_itemsToIndexPaths removeAllObjects];
+  [_indexPathsToItems removeAllObjects];
+  [_itemsToAttributes removeAllObjects];
+  
+  [self setSubviews: [NSArray array]];
+  [_collectionViewLayout prepareLayout];
+
+  // destroy maps...
+  DESTROY(_indexPathsToItems);
+  DESTROY(_itemsToIndexPaths);
+  DESTROY(_itemsToAttributes);
+  
+  // reallocate the maps...
+  _itemsToAttributes = RETAIN([NSMapTable strongToStrongObjectsMapTable]);
+  _itemsToIndexPaths = RETAIN([NSMapTable strongToStrongObjectsMapTable]);
+  _indexPathsToItems = RETAIN([NSMapTable strongToStrongObjectsMapTable]);
+}
+
+- (void) _setFrameWithoutTile: (NSRect)f
```

Comment:
Is this needed as a separate method? I only did see one call and there it could be inlined.

Reply:
Inlined.